README.md
# marketeR ![release](https://img.shields.io/badge/release-v0.5-blue.svg)![License](http://img.shields.io/badge/license-GPL%20%28%3E=%202%29-brightgreen.svg?style=flat)

![marketer_logo](img/marketer_logo.png)

## A Web Analytics Toolbox for Digital Marketers

The `marketeR` package provides a web analytics toolbox for digital marketers using Google Analytics.

Main features:

* Automatic reports
* Week KPIs assessment
* KPIs forecasts

With marketeR, navigate the ocean of web data with the speed of a swordfish :fish: !

## Table of contents

* [Installation](#installation)
* [Authentification](#authentification)
* [Documentation](#documentation)
* [Terms & Conditions](#terms--conditions)

## Installation

Open a new R session, and run the following code:

```
install_github("naileakim/marketeR")
library("marketeR")
```

Note: If you get the error `Error: could not find function "install_github"`, you might need to install / load the package `devtools` first:

```
install.packages("devtools")
library("devtools")
```

## Authentification

### Getting your Google Analytics API credentials

The marketeR package is using the [Google Analytics Reporting API V4](https://developers.google.com/analytics/devguides/reporting/core/v4/) to get the data you need, and require you to be authentified with an access token. To start using marketeR, please follow the instructions below:

- Login with your Google Analytics account
- Go to [Google Developers Console](https://console.developers.google.com/project).
- Click on `Select a project`, then on `Create project`.
- Click on the `API Manager` tab in the sidebar, search for `Analytics Reporting API V4`. After clicking on it, click on `Enable`.
- Click on the `Credentials` tab in the sidebar,  then on `Create credentials`, select  the `Oauth 2.0 client ID` option, check the `Other` box, give a `Name` and click on `Create`.
- You should now be able to get your `Client ID` and `Client secret`.

### Accessing the Google Analytics API from R

Open a new R session, and run the following code with your credentials:

```
client.id <- "PASTE HERE YOUR CLIENT ID"
client.secret <- "PASTE HERE YOUR CLIENT SECRET"
```

Create your token object, then validate it:

```
token <- Auth(client.id, client.secret)
ValidateToken(token)
```

You can also save your token for future sessions:

```
save(token, file="./token_file")
```

So that in future sessions, you don't need to create a new token, you can just load it:

```
load("./token_file")
ValidateToken(token)
```

## Documentation

### Automatic reports

#### Description

The `report` function returns a report presenting your website KPIs during the last month, and their evolution over a one year period.

#### Usage

```
report(website)
```

#### Arguments

* **website:** The unique Google Analytics table ID of the form ga:XXXXXXX, where XXXXXXX is the Analytics view (profile) ID for which the query will retrieve the data.

#### Example

```
report(website = "ga:XXXXXXX")
```
![report](img/report.png)

#### Note

The output file will be generated in your working directory. Don't know where is your working directory ? Type `getwd()` in the R console.

### Week KPIs assessment

#### Description

The `weekperf` function returns a plot designed to assess your website KPIs during the past 7 days.

#### Usage

```
weekperf(kpi, website, export)
```

#### Arguments

* **kpi:** A Google Analytics metric, such as ga:metrics.
* **website:** The unique Google Analytics table ID of the form ga:XXXXXXX, where XXXXXXX is the Analytics view (profile) ID for which the query will retrieve the data.
* **export:** If the export option is set as "TRUE", both raw data & graphics will be exported in the current working directory.

#### Example

```
weekperf(kpi = "ga:sessions", website = "ga:XXXXXXX", export = TRUE)
```

![weekperf](img/weekperf.png)

#### Note

The triangles are representing the results of the past 7 days. Their color may vary according to the mean (green is > mean, red is < mean). The mean is represented by the letter `m`.

### KPIs forecasts

#### Description

The `forecast` function makes forecasts for your website KPIs for the next 3 months, using time-series.

#### Usage

```
forecast(kpi, website, export)
```

#### Arguments
* **kpi:** A Google Analytics metric, such as ga:metrics.
* **website:** The unique Google Analytics table ID of the form ga:XXXXXXX, where XXXXXXX is the Analytics view (profile) ID for which the query will retrieve the data.
* **export:** If the export option is set as "TRUE", both raw data & graphics will be exported in the current working directory.

#### Example

```
forecast(kpi = "ga:sessions", website = "ga:XXXXXXX", export = FALSE)
```

![forecast](img/forecast.png)

#### Note

The black part is the past website traffic data; the blue part is the graphical representation of the forecast.

### Terms & Conditions

### Author

Félix MIKAELIAN <felix.mikaelian@essec.edu> :fr:

### Dependencies

* RGoogleAnalytics
* forecast
* lubridate
* timeDate
* zoo
* plyr
* dplyr
* grid
* scales
* ggplot2
* ggthemes
* rmarkdown
* knitr

### Copyrights & licenses

* The marketeR package is under license GPL (>= 2).
* The marketeR logo is using the *"Enriqueta"* font by [FontFuror](http://www.fontfuror.com/).
* The marketeR logo is using the icon *"Swordfish"* by [Catalina Montes](http://www.pezglobo.cl/).

### Contributing

I would love ideas to make marketeR better, so feel free to contribute to this exciting project :v: !


NEWS.md
## NEWS

### marketeR 0.5 (2016-05-01)

- Functions have been renamed:
	- AutoReport() -> report()
	- WeekSummary() —> weekperf()
	- PerformanceForecast() -> forecast()
	
- Functions now take less and easier arguments.

- The FacebookInsightsDashboard() function was removed (out of package scope).

- Updated README.md documentation.

- Note : Developement will only continue on GitHub (not on CRAN anymore).

*Bug fixes*

### marketeR 0.1.1 (2015-10-21)

*Bug fixes*

- AutoReport() function : The output html file is now appearing in the working directory.

- WeekSummary() function : The output plot is now showing in the R session.

### marketeR 0.1.0 (2015-09-08)

- Initial release on GitHub.

- Initial release on CRAN.

R/forecast.R
#-------------------------------------------------------------------------------
# forecast.R (marketeR)
#-------------------------------------------------------------------------------

#-------------------------------------------------------------------------------
#' @title KPIs forecasts
#' @description The \code{forecast} function makes forecasts for your website KPIs
#' for the next 3 months, using time-series.
#' @param kpi A Google Analytics metric, such as \code{ga:metrics}.
#' @param website The unique Google Analytics table ID of the form 
#' \code{ga:XXXXXXX}, where \code{XXXXXXX} is the Analytics view (profile) ID 
#' for which the query will retrieve the data.
#' @param export If the export option is set as "TRUE", both raw data & graphics 
#' will be exported in the current working directory.
#' @note The black part is the past website traffic data; the blue part is the
#' graphical representation of the forecast.
#' @examples \dontrun{
#' forecast(kpi = "ga:sessions", website = "ga:XXXXXXX", export = FALSE)}
#' @export
#-------------------------------------------------------------------------------

forecast <- function(kpi, website, export = FALSE) {

  #-----------------------------------------------------------------------------
  # Get past.data from Google Analytics Core Reporting API
  #-----------------------------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(72))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = kpi,
                     dimensions = "ga:yearmonth",
                     table.id = website
                     )

  ga.query <- QueryBuilder(query.list)
  past.data <- GetReportData(ga.query, token)

  #-----------------------------------------------------------------------------
  # Clean past.data
  #-----------------------------------------------------------------------------

  past.data$yearmonth <- as.Date(as.yearmon(past.data$yearmonth, "%Y%m"))

  #-----------------------------------------------------------------------------
  # Turn past.data into an exploitable time-serie
  #-----------------------------------------------------------------------------

  ts.past.data <- ts(past.data[, -1],
                     start = c(as.numeric(substring(as.character(timeFirstDayInMonth(Sys.Date() %m-% months(72))), 1, 4)),
                           as.numeric(substring(as.character(timeFirstDayInMonth(Sys.Date() %m-% months(72))), 6, 7))),
                     end = c(as.numeric(substring(as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))), 1, 4)),
                         as.numeric(substring(as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))), 6, 7))),
                     frequency = 12)

  #-----------------------------------------------------------------------------
  # Make predictions.data using the auto-ARIMA model
  #-----------------------------------------------------------------------------

  fit <- auto.arima(ts.past.data)
  predictions.data <- forecast::forecast(fit, 3)
  
  #-----------------------------------------------------------------------------
  # Create plot of predictions.data
  #-----------------------------------------------------------------------------
  
  plot(predictions.data)
  
  #-----------------------------------------------------------------------------
  # Export data
  #-----------------------------------------------------------------------------

  if (export == TRUE) {
    write.csv(predictions.data, "forecast_data.csv")
    png("forecast_plot.png")
    plot(predictions.data)
    dev.off()
    }

}


R/settings.R
#-------------------------------------------------------------------------------
# settings.R (marketeR)
#-------------------------------------------------------------------------------

#-------------------------------------------------------------------------------
#' @import RGoogleAnalytics forecast zoo dplyr ggplot2 grid scales rmarkdown 
#' knitr ggthemes
#' @importFrom lubridate %m-%
#' @importFrom timeDate timeFirstDayInMonth timeLastDayInMonth
#' @importFrom plyr ddply
#' @importFrom grDevices dev.off png
#' @importFrom graphics plot
#' @importFrom stats setNames ts
#-------------------------------------------------------------------------------

globalVariables(c("token",
                  "comma",
                  'sessions',
                  'socialNetwork',
                  'yearMonth',
                  'deviceCategory',
                  'percentages',
                  'percentNewsessions',
                  'avgSessionDuration',
                  'bounceRate',
                  'channelGrouping',
                  'userGender',
                  'userAgeBracket',
                  'dayOfWeek',
                  'sessions',
                  'V2',
                  'dates',
                  'new_dates'
                  )
                )

R/report.R
#-------------------------------------------------------------------------------
# report.R (marketeR)
#-------------------------------------------------------------------------------

#-------------------------------------------------------------------------------
#' @title Automatic reports
#' @description The \code{report} function returns a report presenting your
#' website KPIs during the last month, and their evolution over a one year period.
#' @param website The unique Google Analytics table ID of the form 
#' \code{ga:XXXXXXX}, where \code{XXXXXXX} is the Analytics view (profile) ID 
#' for which the query will retrieve the data.
#' @note The output file will be generated in your working directory. Don't know
#'  where is your working directory ? Type `getwd()` in the R console.
#' @examples \dontrun{
#' report(website = "ga:XXXXXXX")}
#' @export
#-------------------------------------------------------------------------------

report <- function(website) {

  #-----------------------------------------------------------------------------
  # Get report data from Google Analytics Core Reporting API
  #-----------------------------------------------------------------------------

  # basicMetrics.data ----------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(12))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions, ga:percentNewsessions,
                                ga:avgSessionDuration, ga:bounceRate",
                     dimensions = "ga:yearMonth",
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  basicMetrics.data <- GetReportData(ga.query, token)

  # channelGrouping.data -------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(12))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions",
                     dimensions = "ga:yearMonth, ga:channelGrouping",
                     sort = "-ga:sessions",
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  channelGrouping.data <- GetReportData(ga.query, token)

  # socialNetwork.data ---------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(1))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions",
                     dimensions = "ga:yearMonth, ga:socialNetwork",
                     sort = "-ga:sessions",
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  socialNetwork.data <- GetReportData(ga.query, token)

  # keyword.data ---------------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(1))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions",
                     dimensions = "ga:yearMonth, ga:keyword",
                     sort = "-ga:sessions",
                     max.results = 10,
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  keyword.data <- GetReportData(ga.query, token)

  # deviceCategory.data --------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(12))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions",
                     dimensions = "ga:yearMonth, ga:deviceCategory",
                     sort = "-ga:sessions",
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  deviceCategory.data <- GetReportData(ga.query, token)

  # topContent.data ------------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(1))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions",
                     dimensions = "ga:yearMonth, ga:pagePath",
                     sort = "-ga:sessions",
                     max.results = 10,
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  topContent.data <- GetReportData(ga.query, token)

  # country.data ---------------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(1))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions",
                     dimensions = "ga:yearMonth, ga:country",
                     sort = "-ga:sessions",
                     max.results = 10,
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  country.data <- GetReportData(ga.query, token)

  # userGender.data ------------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(12))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions",
                     dimensions = "ga:yearMonth, ga:userGender",
                     sort = "-ga:sessions",
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  userGender.data <- GetReportData(ga.query, token)

  # userAgeBracket.data --------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(12))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:sessions",
                     dimensions = "ga:yearMonth, ga:userAgeBracket",
                     sort = "-ga:sessions",
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  userAgeBracket.data <- GetReportData(ga.query, token)

  # newUsers.data --------------------------------------------------------------

  query.list <- Init(start.date = as.character(timeFirstDayInMonth(Sys.Date() %m-% months(12))),
                     end.date = as.character(timeLastDayInMonth(Sys.Date() %m-% months(1))),
                     metrics = "ga:newUsers",
                     dimensions = "ga:yearMonth",
                     table.id = website)

  ga.query <- QueryBuilder(query.list)
  newUsers.data <- GetReportData(ga.query, token)

  #-----------------------------------------------------------------------------
  # Clean report data
  #-----------------------------------------------------------------------------

  # Change the report data date format -----------------------------------------

  yearmonFormat <- function (x) {
    x <- as.Date(as.yearmon(x, "%Y%m"))
  }

  basicMetrics.data$yearMonth <- yearmonFormat(basicMetrics.data$yearMonth)
  channelGrouping.data$yearMonth <- yearmonFormat(channelGrouping.data$yearMonth)
  socialNetwork.data$yearMonth <- yearmonFormat(socialNetwork.data$yearMonth)
  keyword.data$yearMonth <- yearmonFormat(keyword.data$yearMonth)
  deviceCategory.data$yearMonth <- yearmonFormat(deviceCategory.data$yearMonth)
  topContent.data$yearMonth <- yearmonFormat(topContent.data$yearMonth)
  country.data$yearMonth <- yearmonFormat(country.data$yearMonth)
  userGender.data$yearMonth <- yearmonFormat(userGender.data$yearMonth)
  userAgeBracket.data$yearMonth <- yearmonFormat(userAgeBracket.data$yearMonth)

  # Add percentages ------------------------------------------------------------

  channelGrouping.data <- ddply(channelGrouping.data,
                                "yearMonth",
                                transform,
                                percentages = sessions / sum(sessions))

  deviceCategory.data <- ddply(deviceCategory.data,
                               "yearMonth",
                               transform,
                               percentages = sessions / sum(sessions))

  userGender.data <- ddply(userGender.data,
                           "yearMonth",
                           transform,
                           percentages = sessions / sum(sessions))

  userAgeBracket.data <- ddply(userAgeBracket.data,
                               "yearMonth",
                               transform,
                               percentages = sessions / sum(sessions))

  # Prepare socialNetwork.data data for making a ggplot2 pie chart -------------

  others = data.frame(yearMonth = socialNetwork.data$yearMonth[1],
                      socialNetwork = "Others",
                      sessions = sum(socialNetwork.data$sessions))

  socialNetwork.data <-
    socialNetwork.data %>%
    slice(1:5)

  others$sessions = others$sessions-sum(socialNetwork.data$sessions)
  socialNetwork.data <- bind_rows(socialNetwork.data, others)

  socialNetwork.data <-
    socialNetwork.data %>%
    dplyr::select(socialNetwork, sessions)

  #-----------------------------------------------------------------------------
  # Preparing data for the Executive Summary
  #-----------------------------------------------------------------------------

  ## sessions ------------------------------------------------------------------

  sessions.currentMonth <- as.numeric(basicMetrics.data$sessions[nrow(basicMetrics.data)])
  sessions.lastMonth <- as.numeric(basicMetrics.data$sessions[nrow(basicMetrics.data)-1])
  sessions.relativeChange <- ((sessions.currentMonth/sessions.lastMonth)-1)*100

  ## percentNewsessions --------------------------------------------------------

  percentNewsessions.currentMonth <- as.numeric(basicMetrics.data$percentNewsessions[nrow(basicMetrics.data)])
  percentNewsessions.lastMonth <- as.numeric(basicMetrics.data$percentNewsessions[nrow(basicMetrics.data)-1])
  percentNewsessions.relativeChange <- ((percentNewsessions.currentMonth/percentNewsessions.lastMonth)-1)*100

  ## avgSessionDuration --------------------------------------------------------

  avgSessionDuration.currentMonth <- as.numeric(basicMetrics.data$avgSessionDuration[nrow(basicMetrics.data)])
  avgSessionDuration.lastMonth <- as.numeric(basicMetrics.data$avgSessionDuration[nrow(basicMetrics.data)-1])
  avgSessionDuration.relativeChange <- ((avgSessionDuration.currentMonth/avgSessionDuration.lastMonth)-1)*100

  ## bounceRate ----------------------------------------------------------------

  bounceRate.currentMonth <- as.numeric(basicMetrics.data$bounceRate[nrow(basicMetrics.data)])
  bounceRate.lastMonth <- as.numeric(basicMetrics.data$bounceRate[nrow(basicMetrics.data)-1])
  bounceRate.relativeChange <- ((bounceRate.currentMonth/bounceRate.lastMonth)-1)*100

  ## mobileRate
  deviceCategory.currentMonth <-
    deviceCategory.data %>%
    filter(yearMonth == basicMetrics.data$yearMonth[nrow(basicMetrics.data)]) %>%
    select(deviceCategory, percentages) %>%
    filter(deviceCategory == "mobile")

  deviceCategory.currentMonth$percentages <- as.numeric(substr(deviceCategory.currentMonth$percentages, 0 , nchar(deviceCategory.currentMonth$percentages) - 1))

  deviceCategory.lastMonth <-
    deviceCategory.data %>%
    filter(yearMonth == basicMetrics.data$yearMonth[nrow(basicMetrics.data)-1]) %>%
    select(deviceCategory, percentages) %>%
    filter(deviceCategory == "mobile")

  deviceCategory.lastMonth$percentages <- as.numeric(substr(deviceCategory.lastMonth$percentages, 0 , nchar(deviceCategory.lastMonth$percentages) - 1))

  deviceCategory.relativeChange <- ((deviceCategory.currentMonth$percentages/deviceCategory.lastMonth$percentages)-1)*100

  #-----------------------------------------------------------------------------
  # Create plots of report data
  #-----------------------------------------------------------------------------

  # sessions.plot --------------------------------------------------------------

  sessions.plot <- ggplot(basicMetrics.data, aes(x = yearMonth, y = sessions)) +
    ggtitle("Sessions") +
    geom_line(size = 0.5, color = "red") +
    geom_point(size = 2.5, shape = 21, fill = "white") +
    scale_x_date(labels = date_format("%b %Y"), breaks = date_breaks("month")) +
    scale_y_continuous(labels = comma, limits=c(0, max(basicMetrics.data$sessions))) +
    theme_economist() +
    scale_colour_economist() +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.text.x  = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10))

  # percentNewsessions.plot ----------------------------------------------------

  percentNewsessions.plot <-
    ggplot(basicMetrics.data, aes(x = yearMonth, y = percentNewsessions/100)) +
    ggtitle("Percent of New Sessions") +
    geom_line(size = 0.5, color = "red") +
    geom_point(size = 2.5, shape = 21, fill = "white") +
    scale_x_date(labels = date_format("%b %Y"), breaks = date_breaks("month")) +
    scale_y_continuous(labels  = percent_format(), limits=c(0, max(basicMetrics.data$percentNewsessions/100))) +
    theme_economist() +
    scale_colour_economist() +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.text.x  = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10))

  # avgSessionDuration.plot ----------------------------------------------------

  avgSessionDuration.plot <-
    ggplot(basicMetrics.data, aes(x = yearMonth, y = avgSessionDuration)) +
    ggtitle("Average Session Duration (in seconds)") +
    geom_line(size = 0.5, color = "red") +
    geom_point(size = 2.5, shape = 21, fill = "white") +
    scale_x_date(labels = date_format("%b %Y"), breaks = date_breaks("month")) +
    scale_y_continuous(labels = comma, limits=c(0, max(basicMetrics.data$avgSessionDuration))) +
    theme_economist() +
    scale_colour_economist() +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.text.x  = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10))

  # bounceRate.plot ----------------------------------------------------

  bounceRate.plot <-
    ggplot(basicMetrics.data, aes(x = yearMonth, y = bounceRate/100)) +
    ggtitle(" Bounce Rate (in percent)") +
    geom_line(size = 0.5, color = "red") +
    geom_point(size = 2.5, shape = 21, fill = "white") +
    scale_x_date(labels = date_format("%b %Y"), breaks = date_breaks("month")) +
    scale_y_continuous(labels  = percent_format(), limits=c(0, max(basicMetrics.data$bounceRate/100))) +
    theme_economist() +
    scale_colour_economist() +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.text.x  = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10))

  # channelGrouping.plot -------------------------------------------------------

  channelGrouping.plot <-
    ggplot(channelGrouping.data, aes(x = yearMonth, y = percentages)) +
    geom_bar(aes(fill = channelGrouping), stat = "identity") +
    scale_x_date(labels = date_format("%b %Y"), breaks = date_breaks("month")) +
    scale_y_continuous(labels = percent_format()) +
    theme_economist() +
    scale_colour_economist() +
    scale_fill_brewer(palette = "YlGnBu") +
    guides(fill=guide_legend(title = NULL)) +
    theme(legend.text=element_text(size=10),
          axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.text.x  = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10))

  # socialNetwork.plot ---------------------------------------------------------

  socialNetwork.plot <-
    ggplot(socialNetwork.data, aes(x = "", y = sessions, fill = socialNetwork)) +
    geom_bar(width = 1, stat = "identity") +
    coord_polar("y") +
    theme_economist() +
    scale_fill_brewer(palette = "Pastel2") +
    guides(fill=guide_legend(title = NULL)) +
    geom_text(
      aes(y = sessions/3 + c(0, cumsum(sessions)[-length(sessions)])),
      label = paste(round(socialNetwork.data$sessions / sum(socialNetwork.data$sessions) * 100, 0), "%", sep=""),
      size=8,
      color = "white") +
    theme(legend.position = "right",
          axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          panel.border = element_blank(),
          panel.grid = element_blank(),
          axis.ticks = element_blank(),
          axis.text.x = element_blank(),
          axis.text.y = element_blank())

  # deviceCategory.plot --------------------------------------------------------

  deviceCategory.plot <-
    ggplot(deviceCategory.data, aes(x = yearMonth, y = percentages)) +
    geom_bar(aes(fill = deviceCategory), breaks = date_breaks("month")) +
    scale_x_date(labels = date_format("%b %Y"), breaks = 0:12) +
    scale_y_continuous(labels = percent_format()) +
    theme_economist() +
    scale_colour_economist() +
    scale_fill_brewer(palette = "RdPu") +
    guides(fill=guide_legend(title = NULL)) +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.text.x  = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10))

  # userGender.plot ------------------------------------------------------------

  userGender.plot <-
    ggplot(userGender.data, aes(x = yearMonth, y = percentages)) +
    geom_bar(aes(fill = userGender), stat = "identity") +
    scale_x_date(labels = date_format("%b %Y"), breaks = date_breaks("month")) +
    scale_y_continuous(labels = percent_format()) +
    theme_economist() +
    scale_colour_economist() +
    scale_fill_brewer(palette = "Pastel1") +
    guides(fill=guide_legend(title = NULL)) +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.text.x  = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10))

  # userAgeBracket.plot --------------------------------------------------------

  userAgeBracket.plot <-
    ggplot(userAgeBracket.data, aes(x = yearMonth, y = percentages)) +
    geom_bar(aes(fill = userAgeBracket), stat = "identity") +
    scale_x_date(labels = date_format("%b %Y"), breaks = date_breaks("month")) +
    scale_y_continuous(labels = percent_format()) +
    theme_economist() +
    scale_colour_economist() +
    scale_fill_brewer(palette = "Pastel1") +
    guides(fill=guide_legend(title = NULL)) +
    theme(axis.title.x = element_blank(),
          axis.title.y = element_blank(),
          axis.text.x  = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 10))

  #-----------------------------------------------------------------------------
  # Multiple plot function from cookbook-r.com by Winston Chang
  #-----------------------------------------------------------------------------

  multiplot <- function(..., plotlist=NULL, file, cols=1, layout=NULL) {

    # Make a list from the ... arguments and plotlist
    plots <- c(list(...), plotlist)

    numPlots = length(plots)

    # If layout is NULL, then use 'cols' to determine layout
    if (is.null(layout)) {
      # Make the panel
      # ncol: Number of columns of plots
      # nrow: Number of rows needed, calculated from # of cols
      layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),
                       ncol = cols, nrow = ceiling(numPlots/cols))
    }

    if (numPlots==1) {
      print(plots[[1]])

    } else {
      # Set up the page
      grid.newpage()
      pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))

      # Make each plot, in the correct location
      for (i in 1:numPlots) {
        # Get the i,j matrix positions of the regions that contain this subplot
        matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))

        print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                        layout.pos.col = matchidx$col))
      }
    }
  }

  #-----------------------------------------------------------------------------
  # Find then render the report.rmd file
  #-----------------------------------------------------------------------------

  rmdDir <- system.file("report", "report.Rmd", package = "marketeR")
  if (rmdDir == "") {
    stop("Could not find given directory. Try re-installing marketeR.", call. = FALSE)
  }

  rmarkdown::render(rmdDir, output_file = '~/report.html')

}
